build/source/engine/computFlux.f90

```diff
@@ -0,0 +1,854 @@
+    else
+     mLayerdTheta_dTk(iLayer)        = 0._dp
+    endif
+   case default; err=40; message=trim(message)//"cannot identify the layer as snow or soil"; return
+  endselect
+ end do  ! (looping through snow+soil layers)
+
+ ! * compute the matric head associated with liquid water
+ do iSoil=1,nSoil  ! loop through soil layers
+
+  ! - compute derivative in total water content w.r.t. total water matric potential (m-1)
+  dVolTot_dPsi0(iSoil) = dTheta_dPsi(mLayerMatricHeadTrial(iSoil),vGn_alpha,theta_res,theta_sat,vGn_n,vGn_m)  ! valid for both frozen and unfrozen conditions
+
+  ! ** partially frozen soil
+  if(mLayerVolFracIceTrial(nSnow+iSoil) > verySmall .and. mLayerMatricHeadTrial(iSoil) < 0._dp)then  ! check that ice exists and that the soil is unsaturated
+   ! - compute effective saturation
+   ! NOTE: include ice content as part of the solid porosity - major effect of ice is to reduce the pore size; ensure that effSat=1 at saturation
+   ! (from Zhao et al., J. Hydrol., 1997: Numerical analysis of simultaneous heat and mass transfer...)
+   xNum   = mLayerVolFracLiqTrial(nSnow+iSoil) - theta_res
+   xDen   = theta_sat - mLayerVolFracIceTrial(nSnow+iSoil) - theta_res
+   effSat = xNum/xDen          ! effective saturation
+   ! - matric head associated with liquid water
+   mLayerMatricHeadLiq(iSoil) = matricHead(effSat,vGn_alpha,0._dp,1._dp,vGn_n,vGn_m)  ! argument is effective saturation, so theta_res=0 and theta_sat=1
+   ! - compute derivative in the liquid water matric potential w.r.t. the total water matric potential
+   dPsiLiq_dEffSat      = dPsi_dTheta(effSat,vGn_alpha,0._dp,1._dp,vGn_n,vGn_m) ! derivative in liquid water matric potential w.r.t. effective saturation (m)
+   dEffSat_dVolTot      = xNum/(xDen**2._dp) ! derivative in effective saturation w.r.t. total water content (-)
+   dPsiLiq_dPsi0(iSoil) = dVolTot_dPsi0(iSoil)*dPsiLiq_dEffSat*dEffSat_dVolTot
+   ! compute the derivative in the liquid water matric potential w.r.t. temperature (m K-1)
+   dEffSat_dTemp        = -mLayerdTheta_dTk(nSnow+iSoil)*xNum/(xDen**2._dp) + mLayerdTheta_dTk(nSnow+iSoil)/xDen
+   dPsiLiq_dTemp(iSoil) = dPsiLiq_dEffSat*dEffSat_dTemp
+  ! ** unfrozen soil
+  else   ! (no ice)
+   dPsiLiq_dPsi0(iSoil)       = 1._dp  ! derivative=1 because values are identical
+   dPsiLiq_dTemp(iSoil)       = 0._dp  ! derivative=0 because no impact of temperature for unfrozen conditions
+   mLayerMatricHeadLiq(iSoil) = mLayerMatricHeadTrial(iSoil) ! liquid water matric potential is equal to the total water matic potential when there is no ice
+  endif  ! (if ice exists)
+
+ end do  ! (looping through soil layers)
+
+ ! initialize liquid water fluxes throughout the snow and soil domains
+ ! NOTE: used in the energy routines, which is called before the hydrology routines
+ if(firstFluxCall)then
+  if(nSnow > 0)&
+  iLayerLiqFluxSnow(0:nSnow) = 0._dp
+  iLayerLiqFluxSoil(0:nSoil) = 0._dp
+ endif
```

Comment:
Is there a missing `endif`? Or so you get around that with the line continuation. If so, may be clearer to write:

```Fortran
if(firstFluxCall)then
 if(nSnow > 0) iLayerLiqFluxSnow(0:nSnow) = 0._dp
 iLayerLiqFluxSoil(0:nSoil) = 0._dp
endif
```